### DISCUSSION TITLE
Feature Request: Allow opening read-only database from read-only filesystem

### DISCUSSION THREAD

**Discussion by znation** [duckdb/duckdb#8406] [Ideas]
Currently, DuckDB seems to error when opening a DB from a read-only filesystem, even though I am specifying that I only want to read from it:
```
➜ duckdb -readonly
v0.8.1 6536a77232
Enter ".help" for usage hints.
D .open duckdb.db
Error: unable to open database "duckdb.db": IO Error: Cannot open file "duckdb.db": Read-only file system
D 
```

It would be nice to be able to read from databases stored on read-only filesystems. A particular use case I have in mind is to use `xet mount` from https://github.com/xetdata/pyxet, but I could imagine this being useful for other use cases (removable media, especially in cases where you really don't want to accidentally write to it).

**Comment by Mause** [comment]
Can you try
```sh
duckdb duckdb.db -readonly
```

The way you tried was creating a read-only in memory database, then opening another database in read mode

**Reply by znation** [reply]
Ah, I may have simply been confused by the syntax. I thought -readonly would apply to all opened databases.

**Reply by znation** [reply]
With that command syntax I get a different error but the core problem still seems to be the read-only filesystem:
```
➜ duckdb duckdb/duckdb.db -readonly
Error: unable to open database "duckdb/duckdb.db": IO Error: Could not set lock on file "duckdb/duckdb.db": Operation not supported
```
